File: Dart/Sorting/insertion_sort.dart
//Title:Insertion sort
// Author:ShivamVerma
// Email:shivamthegreat.sv@gmail.com


void insertionSort(List arr, int n)  
{  
    int i, key, j;  
    for (i = 1; i < n; i++) 
    {  
        key = arr[i];  
        j = i - 1;  
  
        /* Move elements of arr[0..i-1], that are  
        greater than key, to one position ahead  
        of their current position */
        while (j >= 0 && arr[j] > key) 
        {  
            arr[j + 1] = arr[j];  
            j = j - 1;  
        }  
        arr[j + 1] = key;  
    }  
}  

/* Driver code */
int main()  
{  
    List arr = [ 12, 11, 13, 5, 6 ];  
    int n = arr.length;  
  
    insertionSort(arr, n);  

    for (int i = 0; i < n; i++)  
        print("${arr[i]} ");
  
    return 0;  
}  

File: Java/Sorting/InsertionSort.java
public class InsertionSort {  
public static void main(String[] args) {  
    int[] a = {10, 9, 7, 101, 23, 44, 12, 78, 34, 23};  
    for(int k=1; k<10; k++)   
    {  
        int temp = a[k];  
        int j= k-1;  
        while(j>=0 && temp <= a[j])  
        {  
            a[j+1] = a[j];   
            j = j-1;  
        }  
        a[j+1] = temp;  
    }  
    System.out.println("printing sorted elements ...");  
    for(int i=0;i<10;i++)  
    {  
        System.out.println(a[i]);  
    }  
}  
} 